File: java/ordenamiento/insercion/InsercionTest.java
import java.util.Arrays;

public class InsercionTest {
	public static void main(String[] args) {
		int datos[] = new int[]{8,5,3,9,1};		
		System.out.println("Arreglo original: " + Arrays.toString(datos));
		
		for (int recorrido = 1; recorrido < datos.length; recorrido++) {
			int buffer = datos[recorrido];
			for (int indice = recorrido; indice > 0; indice--) {
				if ( datos[indice] < datos[indice - 1] ) { //Intercambio
					buffer = datos[indice];
					datos[indice] = datos[indice - 1];
					datos[indice - 1] = buffer;
				}		
			}
		}
		
		System.out.println("Arreglo ordenado: " + Arrays.toString(datos));	
	}
}

File: python/ordenamiento/insercion/InsercionTest.py
import numpy as np

datos = np.array([8,5,3,9,1])
print ("Arreglo original: " + str(datos))

for recorrido in range(1, datos.size, 1):
	buffer = datos[recorrido] 
	for indice in range(recorrido, 0, -1): 				
		if (datos[indice] < datos[indice - 1]): # Intercambio
			buffer = datos[indice]
			datos[indice] = datos[indice - 1]
			datos[indice - 1] = buffer

print ("Arreglo ordenado: " + str(datos))